modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/MaintenanceRegistrySimpleTest.java

```diff
@@ -139,6 +139,48 @@ public void testMaintenanceTaskReplacement() throws IgniteCheckedException {
         assertEquals(newParams, task.parameters());
     }
 
+    /**
+     * {@link MaintenanceTask} could be updated using remapping function.
+     *
+     * @throws IgniteCheckedException If initialization failed.
+     */
+    @Test
+    public void testMaintenanceTaskUpdate() throws IgniteCheckedException {
+        String name0 = "taskName0";
+        String descr = "description";
+        String oldParams = "oldParams";
+        String newParams = "newParams";
+
+        MaintenanceProcessor proc = new MaintenanceProcessor(initContext(true));
+
+        proc.start();
+
+        assertFalse(proc.isMaintenanceMode());
+
+        MaintenanceTask oldTask = new MaintenanceTask(name0, descr, oldParams);
+
+        proc.registerMaintenanceTask(oldTask);
+
+        MaintenanceTask newTask = new MaintenanceTask(name0, descr, newParams);
+
+        proc.registerMaintenanceTask(newTask, prevTask -> new MaintenanceTask(
```

Comment:
What are we expecting here? it seems that the previous task will be returned, wdyt?